### PR TITLE
add other build tools to cmake toolchains

### PR DESCRIPTION
**Pull request by vchuravy**
In order to  bootstrap LLVM properly I need to provide my own toolchain file currently
since the one we use doesn't note which `nm` etc to use

https://github.com/JuliaPackaging/Yggdrasil/pull/296/files#diff-08724ed72679c754126058e3bfd19c92R76

@staticfloat please review and merge

### CODE REVIEW

File: 0_RootFS/PlatformSupport/bundled/buildsystem_toolchains/cmake_macos_gcc.j2 [hunk 1]
```diff
@@ -17,7 +17,7 @@ set(CMAKE_SYSTEM_FRAMEWORK_PATH
 )
 set(CMAKE_INSTALL_PREFIX $ENV{prefix})
 
-set(CMAKE_C_COMPILER /opt/bin/{{TARGET}-gcc)
+set(CMAKE_C_COMPILER /opt/bin/{{TARGET}}-gcc)
```
Comment (staticfloat) [inline 0_RootFS/PlatformSupport/bundled/buildsystem_toolchains/cmake_macos_gcc.j2]: `/me blushes furiously`

File: 0_RootFS/PlatformSupport/bundled/buildsystem_toolchains/cmake_simple_gcc.j2 [hunk 2]
```diff
@@ -7,6 +7,11 @@ set(CMAKE_INSTALL_PREFIX $ENV{prefix})
 
 set(CMAKE_C_COMPILER /opt/bin/{{TARGET}}-gcc)
 set(CMAKE_CXX_COMPILER /opt/bin/{{TARGET}}-g++)
+set(CMAKE_AR /opt/bin/{{TARGET}}-ar)
+set(CMAKE_LINKER /opt/bin/{{TARGET}}-ld)
+set(CMAKE_NM /opt/bin/{{TARGET}}-nm)
+set(CMAKE_RANLIB /opt/bin/{{TARGET}}-ranlib)
+set(CMAKE_OBJCOPY /opt/bin/{{TARGET}}-objcopy)
```
Comment (staticfloat) [inline 0_RootFS/PlatformSupport/bundled/buildsystem_toolchains/cmake_simple_gcc.j2]: Can you add the appropriate mappings for the other toolchains here as well?

**Comment by giordano** [comment]
While we are at it, does it make sense to set `CMAKE_INSTALL_LIBDIR` to `${libdir}`?

**Comment by staticfloat** [comment]
We should also probably add these tools to the `clang` versions as well.

**Comment by vchuravy** [comment]
This is done from my side

**Pull request merged**